op-node/rollup/derive/data_source.go

```diff
@@ -91,9 +91,15 @@ type DataSourceConfig struct {
 }
 
 // isValidBatchTx returns true if:
-//  1. the transaction has a To() address that matches the batch inbox address, and
-//  2. the transaction has a valid signature from the batcher address
+//  1. the transaction type is any of Legacy, ACL, DynamicFee, or Blob.
+//  2. the transaction has a To() address that matches the batch inbox address, and
+//  3. the transaction has a valid signature from the batcher address
 func isValidBatchTx(tx *types.Transaction, l1Signer types.Signer, batchInboxAddr, batcherAddr common.Address, logger log.Logger) bool {
+	// For now, we want to disallow the SetCodeTx type or any future types.
```

Comment:
We should revisit if this is necessary still. This rule was introduced to save the work of needing to know about the serialization of the SetCodeTx. Since that approach didn't work, not sure if we need to add this extra rule given we did the work for the rebase. My reasoning is to reduce branching logic that needs additional test coverage. An honest batcher just won't submit other tx types

Reply:
OTOH this rule prepares forwards-compatibility, as we have specified it in [this specs update](https://github.com/ethereum-optimism/specs/pull/518). So having this rule in now means we can handle generic future tx type and receipt type handling as a soft fork in the future. I don't see a use case for `SetCode` batcher txs  right now.

Without this rule, I think adding future-proof transaction type handling is harder.

Reply:
[Decision was made](https://github.com/ethereum-optimism/pm/issues/28) to keep this rule in, but also allow Deposit transactions. Updated.

Reply:
I added a test to the blob source tests that a SetCode tx is processed and dropped, that otherwise looks like a legit batcher tx.